spatialdata/_core/spatialdata.py

```diff
@@ -1,25 +1,242 @@
+import hashlib
+import os
+import tempfile
+from functools import singledispatch
 from types import MappingProxyType
-from typing import Any, Mapping, Optional
+from typing import Any, Iterable, Mapping, Optional, Tuple
 
+import numpy as np
+import xarray as xr
+import zarr
 from anndata import AnnData
+from dask.array.core import Array as DaskArray
+from ome_zarr.io import parse_url
+
+from spatialdata._core.elements import Image, Labels, Points, Polygons
+from spatialdata._core.transform import Transform, get_transform
+from spatialdata.utils import are_directories_identical
 
 
 class SpatialData:
     """Spatial data structure."""
 
-    adata: AnnData
-    regions: Mapping[str, Any]
-    images: Mapping[str, Any]
-    points: Optional[AnnData]
+    images: Optional[Mapping[str, Image]] = None
+    labels: Optional[Mapping[str, Labels]] = None
+    points: Optional[Mapping[str, Points]] = None
+    polygons: Optional[Mapping[str, Polygons]] = None
+    tables: Optional[Mapping[str, AnnData]] = None
 
     def __init__(
         self,
-        adata: AnnData,
-        regions: Mapping[str, Any] = MappingProxyType({}),
-        images: Mapping[str, Any] = MappingProxyType({}),
-        points: Optional[AnnData] = None,
+        images: Optional[Mapping[str, Any]] = MappingProxyType({}),
+        labels: Optional[Mapping[str, Any]] = MappingProxyType({}),
+        points: Optional[Mapping[str, AnnData]] = MappingProxyType({}),
+        polygons: Optional[Mapping[str, AnnData]] = MappingProxyType({}),
+        tables: Optional[Mapping[str, AnnData]] = MappingProxyType({}),
```

Comment:
Why are these `Optional`? They have default values.

I think it simplifies the model to say they must be `Mapping`s. E.g. cannot be `None`, and are instead just empty dicts.

```suggestion
        images: Mapping[str, Any] = MappingProxyType({}),
        labels: Mapping[str, Any] = MappingProxyType({}),
        points: Mapping[str, AnnData] = MappingProxyType({}),
        polygons: Mapping[str, AnnData] = MappingProxyType({}),
        tables: Mapping[str, AnnData] = MappingProxyType({}),
```